go/mysql/endtoend/main_test.go

```diff
@@ -41,38 +43,28 @@ var (
 // assertSQLError makes sure we get the right error.
 func assertSQLError(t *testing.T, err error, code int, sqlState string, subtext string, query string) {
 	t.Helper()
+	require.Error(t, err, "was expecting SQLError %v / %v / %v but got no error.", code, sqlState, subtext)
 
-	if err == nil {
-		t.Fatalf("was expecting SQLError %v / %v / %v but got no error.", code, sqlState, subtext)
-	}
 	serr, ok := err.(*mysql.SQLError)
-	if !ok {
-		t.Fatalf("was expecting SQLError %v / %v / %v but got: %v", code, sqlState, subtext, err)
-	}
-	if serr.Num != code {
-		t.Fatalf("was expecting SQLError %v / %v / %v but got code %v", code, sqlState, subtext, serr.Num)
-	}
-	if serr.State != sqlState {
-		t.Fatalf("was expecting SQLError %v / %v / %v but got state %v", code, sqlState, subtext, serr.State)
-	}
+	require.True(t, ok, "was expecting SQLError %v / %v / %v but got: %v", code, sqlState, subtext, err)
+	require.Equal(t, code, serr.Num, "was expecting SQLError %v / %v / %v but got code %v", code, sqlState, subtext, serr.Num)
+	require.Equal(t, sqlState, serr.State, "was expecting SQLError %v / %v / %v but got state %v", code, sqlState, subtext, serr.State)
+
 	if subtext != "" && !strings.Contains(serr.Message, subtext) {
 		t.Fatalf("was expecting SQLError %v / %v / %v but got message %v", code, sqlState, subtext, serr.Message)
 	}
```

Comment:
Here we can use:
```
require.True(t, (subtext == "" && strings.Contains(serr.Message, subtext)), "was expecting SQLError %v / %v / %v but got message %v", code, sqlState, subtext, serr.Message)
```

Reply:
same comment about multiple assertions (`require`s, in this case)

Reply:
@mattlord when you take the negation `&&` will become `||`.

Reply:
@ajm188 I don't think we can use multiple assertions here because it's `||`.